source/service/k3s.rst

```diff
@@ -0,0 +1,952 @@
+          "scheduled_policies": []
+        },
+        {
+          "name": "worker",
+          "cardinality": 1,
+          "vm_template": <<<VM_TEMPLATE_ID>>> ,
+          "vm_template_contents": "ONEGATE_ENABLE=\"YES\"",
+          "parents": [
+            "master"
+          ],
+          "elasticity_policies": [],
+          "scheduled_policies": []
+        }
+      ],
+      "ready_status_gate": true
+    }
+
+Or, use the `OneFlow CLI <https://docs.opennebula.io/stable/management_and_operations/multivm_service_management/appflow_use_cli.html#managing-services>`_.
+
+Instantiate Service
+~~~~~~~~~~~~~~~~~~~
+
+To run the multi-node service:
+
+- Go to **Templates → Services** tab
+- Select the K3s service template
+- Click on the green **+** button at the top and **Instantiate**
+- For each role choose ``YES`` under the **Enable OneGate reporting? (req. for multi-node)**
+- Click on **Instantiate**
+
+.. important::
+
+   Mostly you shouldn't be asked for any inputs, but in case the OpenNebula provides you with a list of all K3s specific contextualization variables per role to customize before instantiation, you can leave all the parameters empty. **DO NOT** set any inputs for the **worker** role, the parameters will be configured automatically from the **master** node.
+
+|image-oneflow-part5|
+
+Follow the service deployment status in **Instances → Services** tab.
+
+Check K3s Cluster Status
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Log into the **master node** over SSH and check the cluster contains all the deployed nodes.
+
+For example:
+
+.. prompt:: text # auto
+
+   # kubectl get nodes -o wide
+   NAME                        STATUS   ROLES                  AGE   VERSION        INTERNAL-IP       EXTERNAL-IP   OS-IMAGE             KERNEL-VERSION   CONTAINER-RUNTIME
+   onek3s-ip-192-168-150-100   Ready    control-plane,master   56s   v1.21.5+k3s2   192.168.150.100   <none>        Alpine Linux v3.14   5.10.82-0-virt   containerd://1.4.11-k3s1
+   onek3s-ip-192-168-150-102   Ready    <none>                 17s   v1.21.5+k3s2   192.168.150.102   <none>        Alpine Linux v3.14   5.10.82-0-virt   containerd://1.4.11-k3s1
+   onek3s-ip-192-168-150-101   Ready    <none>                 18s   v1.21.5+k3s2   192.168.150.101   <none>        Alpine Linux v3.14   5.10.82-0-virt   containerd://1.4.11-k3s1
+
+.. _k3s_remote:
+
+Remote Usage
+------------
+
+To control the K3s cluster remotely, you need to have the ``kubectl`` CLI tool installed on your system (workstation, laptop). Follow the official `installation guide <https://kubernetes.io/docs/tasks/tools/install-kubectl/>`_. When finished, you can validate the correct installation by running:
+
+.. prompt:: text [remote]$ auto
+
+    [remote]$ kubectl --help
+
+You also need to have the configuration with cluster master IP address and access keys. This configuration can be taken from the **master** node from a file ``/etc/rancher/k3s/k3s.yaml`` and put on your remote system into ``~/.kube/config``.
```

Comment:
It should be added here that the user should replace the server IP in the file `/etc/rancher/k3s/k3s.yaml` with the master VM IP, since it is set to `127.0.0.1`.

Reply:
Good catch!